### PR TITLE
Support TRUNCATE TABLE statement

### LINKED ISSUE #8921: Support TRUNCATE TABLE statement
This statement is DML (not DDL) in ANSI SQL. The keyword `TRUNCATE` is a part of reserved words. 

```
<truncate table statement> ::=
  TRUNCATE TABLE <target table>  [ <identity column restart option>  ]

<identity column restart option> ::=
    CONTINUE IDENTITY
  | RESTART IDENTITY
```

Output of other databases:
MySQL
```
mysql> truncate table t1;
Query OK, 0 rows affected (0.01 sec)
```

PostgreSQL
```
postgres=# truncate table test.t1;
TRUNCATE TABLE
```

MS SQL Server
```
mssql> truncate table t1;
OK

Executed in 0 ms
```

Oracle
```
SQL> truncate table t1;
 
Table truncated.
```

Cassandra
```
cqlsh> truncate table test.t1;
cqlsh> 
```

Relates to https://trinodb.slack.com/archives/CFLB9AMBN/p1629328499266100

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes #8921

Question: Should we implement as DML in Trino? Implementing as DDL simplifies some code around planning. 

#

- [x] Run test against MemSQL (Oct 19)

### CODE REVIEW

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java [hunk 1]
```diff
@@ -513,6 +514,14 @@ public void checkCanDropTable(SystemSecurityContext context, CatalogSchemaTableN
         }
     }
 
+    @Override
+    public void checkCanTruncateTable(SystemSecurityContext context, CatalogSchemaTableName table)
+    {
+        if (!checkTablePermission(context, table, DELETE)) {
```
Comment (kokosing) [inline lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java:520]: Please add tests

**Comment by kokosing** [comment]
> Question: Should we implement as DML in Trino? Implementing as DDL simplifies some code around planning.

Do you know how ANSI SQL defines `TRUNCATE`?

**Comment by findepi** [comment]
> Question: Should we implement as DML in Trino? Implementing as DDL simplifies some code around planning.

Sounds more like "DDL" to me (i.e. could follow the same way as DROP TABLE is done, as they are somewhat similar).

**Comment by ebyhr** [comment]
It's defined as DML according to this page. 
https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#_14_10_truncate_table_statement

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 2]
```diff
@@ -901,6 +902,17 @@ public void testRowLevelDelete()
         }
     }
 
+    @Test
+    public void testTruncateTable()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_TRUNCATE));
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_truncate", "AS SELECT * FROM region")) {
+            assertUpdate("TRUNCATE TABLE " + table.getName());
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:911]: As far as I checked MySQL, PostgreSQL and Oracle, they always return 0 in [Statement.executeUpdate](https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Statement.html#executeUpdate(java.lang.String)) method. We could propagate the value, but I feel it's misleading especially in CLI (always TRUNCATE TABLE: 0 rows)

**Comment by martint** [comment]
The syntax matches what's described in the SQL spec, so I remove the `syntax-needs-review` label.

**Pull request merged**